lib/scope.js

```diff
@@ -52,5 +62,64 @@ module.exports = {
     // Plain old waiting. Yeah, I abstracted it into here so only
     // one thing was being called at the end
     if ( options.waitForTimeout ) { await scope.page.waitFor( options.waitForTimeout ); }
-  },
+  },  // Ends afterStep
+
+  getSpecifiedChoice: async function getSpecifiedChoice(scope, {ordinal, label_text, group_label_text, roles}) {
+    /* Returns the elementHandle of the field specified by the arguments
+    * 
+    * @param {string} [ordinal='first'] Which instance of the field on the page.
+    * @param {string} [label_text=''] Text of the label for the field, if it has one.
+    * @param {string} [group_label_text=''] Text of the label for the group of fields to which the field belongs, if there is one.
+    * @param {array} role_types 'radio', 'checkbox', or both. The type of field that the developer could be asking for.
+    * 
+    * @returns instance of ElementHandle or test fails
+    * 
+    * Methodologies proposed:
+    *     * Proposed: First dig all the way down to the label, then dig back out
+    *     * Regected: It's not clear how to keep things in order and filtering them
+    *     that way is more confusing, not less. It also takes more lines of code.
+    *     * Proposed: First collect all elements of expected status.
+    *     * Rejected: That ruins getting the correct ordinal
```

Comment:
This doesn't seem to list the methodology you ended up using here. I think it would be awesome to have a summary of how this function works, because it is so complicated.